pkg/v1/tkg/client/cluster.go

```diff
@@ -468,6 +470,10 @@ func (c *TkgClient) createPacificCluster(options *CreateClusterOptions, waitForC
 		return nil
 	}
 
+	if options.IsInputFileClusterClassBased {
+		return c.waitForClusterCreation(clusterClient, options, options.IsInputFileClusterClassBased)
+	}
+
 	log.V(3).Infof("Waiting for the Tanzu Kubernetes Cluster service for vSphere workload cluster\n")
 
 	if err := clusterClient.WaitForPacificCluster(clusterName, namespace); err != nil {
```

Comment:
Is the if case supposed to replace this case? iow should L479 be in the else clause?

Reply:
Yes, that makes sense keep existing logic in else case and validate the error and return, I will update the code.